crates/pop-parachains/src/call/metadata/params.rs

```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::errors::Error;
+use pop_common::format_type;
+use scale_info::{form::PortableForm, Field, PortableRegistry, TypeDef};
+use subxt::{Metadata, OnlineClient, SubstrateConfig};
+
+/// Describes a parameter of an extrinsic.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Param {
+	/// The name of the parameter.
+	pub name: String,
+	/// The type of the parameter.
+	pub type_name: String,
+	/// Nested parameters for composite, variants types or tuples.
+	pub sub_params: Vec<Param>,
+	/// Indicates if the parameter is optional (`Option<T>`).
+	pub is_optional: bool,
+	/// Indicates if the parameter is a Tuple.
+	pub is_tuple: bool,
+	/// Indicates if the parameter is a Variant.
+	pub is_variant: bool,
+}
+
+/// Transforms a metadata field into its `Param` representation.
+///
+/// # Arguments
+/// * `client`: Reference to an `OnlineClient` connected to the blockchain.
```

Comment:
Please go through your PR and see where else the parameters can be explained rather than saying the type and whether it has a reference

Reply:
As above, improved here: https://github.com/r0gue-io/pop-cli/pull/316/commits/e70bc7a6dbefd8e37165f102654d012605c51a37